util/downloader/downloader.go

```diff
@@ -106,13 +104,25 @@ func (d *Downloader) download(ctx context.Context) {
 		_ = out.Close()
 	}()
 
-	if err := d.validateExistingFile(out, &stats); err != nil {
-		d.handleError(err)
+	d.statsCh <- stats
 
-		return
+	var wg sync.WaitGroup
+	for _, c := range d.chunks {
+		wg.Add(1)
+		go func(c *chunk) {
+			defer wg.Done()
+			err := d.downloadChunkWithContext(ctx, out, c, stats.TotalSize)
+			if err != nil {
+				d.handleError(err)
+
+				return
+			}
+		}(c)
 	}
 
-	if err := d.downloadFile(ctx, out, &stats); err != nil {
+	wg.Wait()
```

Comment:
If one chunk got an error, do we have wait here?

Reply:
@b00f don't wait here return error for handling then send `wg.Done()` signal.

![image](https://github.com/user-attachments/assets/c2a18f47-3676-45c6-a379-35e495cdaf21)